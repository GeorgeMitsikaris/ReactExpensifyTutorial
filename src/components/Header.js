import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
        <header>
            <h1>Expensify</h1><br/>
            <NavLink to="/" exact={true} activeClassName="is-active">Dashboard</NavLink><br/>
            <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br/>
        </header>
    </div>
)

export default Header;