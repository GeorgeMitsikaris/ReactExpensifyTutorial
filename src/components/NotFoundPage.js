import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        <div>Page Not Found</div>
        <Link to="/">Go Home</Link>
    </div>
)

export default NotFoundPage;