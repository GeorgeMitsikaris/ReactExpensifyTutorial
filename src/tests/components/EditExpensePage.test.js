import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import { expenses, expenseTwo } from '../fixtures/expenses';

let editExpense, startRemoveExpense, history, wrapper;

beforeEach(() => {
     editExpense = jest.fn();
     startRemoveExpense = jest.fn();
     history = {push:jest.fn()};
     wrapper = shallow(<EditExpensePage
          editExpense = {editExpense}
          startRemoveExpense = {startRemoveExpense}
          history = {history}
          expense = {expenseTwo}
     />)
})

test('should render EditExpensePage', () => {
     expect(wrapper).toMatchSnapshot();
})

test('should handle editExense', () => {
     wrapper.find('ExpenseForm').prop('onSubmit')(expenseTwo);
     expect(history.push).toHaveBeenLastCalledWith('/');
     expect(editExpense).toHaveBeenLastCalledWith(expenseTwo.id, expenseTwo);
})

test('should handle startRemoveExpense', () => {
     wrapper.find('button').simulate('click'); 
     expect(history.push).toHaveBeenLastCalledWith('/');
     expect(startRemoveExpense).toHaveBeenLastCalledWith({id: expenseTwo.id})
})