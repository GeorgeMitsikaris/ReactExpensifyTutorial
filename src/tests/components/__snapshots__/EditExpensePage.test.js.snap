// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 100,
        "createdAt": -345600000,
        "description": "description for gas",
        "id": "2",
        "note": "gas",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove Expense
  </button>
</div>
`;
